### PR TITLE
Ratelimit kbfs notifications

### PR DESCRIPTION
**Pull request by cjb**
Here's a quick and dirty ratelimiting of KBFS notifications.

After sending out a few notifications, here's what the `sentNotifications` object looks like:
```js
{ 
  Decrypting: 
   { Starting: { '/private/cjb': Tue Dec 01 2015 16:22:07 GMT-0500 (EST) },
     Finished: { '/private/cjb': Tue Dec 01 2015 16:22:07 GMT-0500 (EST) } },
  Verifying: 
   { Starting: { '/public/chris': Tue Dec 01 2015 16:23:46 GMT-0500 (EST) },
     Finished: { '/public/chris': Tue Dec 01 2015 16:23:46 GMT-0500 (EST) } }
}
```

Style-wise, was splitting out `rateLimitAllowsNotify()` a good idea?

Also includes a fix for Linux runMode==prod.

### CODE REVIEW

**Comment by cjb** [comment]
@keybase/react-hackers PTAL, thanks!

File: react-native/react/constants/platform.native.desktop.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ function findSocketRoot () {
     'linux': `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`
   }
 
+  if (runMode === 'prod') {
```
Comment (chrisnojima) [inline react-native/react/constants/platform.native.desktop.js]: feels better to do 

```
const paths = {
     'darwin': `${process.env.HOME}/Library/Caches/${envedPathOSX[runMode]}/`,
     'linux': runMode === 'prod' ? `${process.env.XDG_RUNTIME_DIR}/keybase/ : `${process.env.XDG_RUNTIME_DIR}/keybase.${runMode}/`
   }
```

or something i think vs mutating the const paths?

File: react-native/react/native/notification-listeners.desktop.js [hunk 2]
```diff
@@ -23,11 +27,43 @@ export default {
       [enums.kbfs.FSStatusCode.error]: 'Errored'
     }[notification.statusCode]
 
-    const pubPriv = notification.publicTopLevelFolder ? '[Public]' : '[Private]'
+    const basedir = notification.filename.split('/')[0]
```
Comment (chrisnojima) [inline react-native/react/native/notification-listeners.desktop.js]: better to use node helpers right?

https://nodejs.org/api/path.html#path_path_parse_pathstring

File: react-native/react/native/notification-listeners.desktop.js [hunk 3]
```diff
@@ -23,11 +27,43 @@ export default {
...
+    let tlf
+    if (notification.publicTopLevelFolder) {
+      // Public filenames look like cjb#public/foo.txt
+      tlf = '/public/' + basedir.replace('#public', '')
```
Comment (chrisnojima) [inline react-native/react/native/notification-listeners.desktop.js]: maybe better to use `/public/${basedir...}`

**Comment by chrisnojima** [comment]
:+1: % comments

**Comment by cjb** [comment]
Thanks!  Made fixes for every comment, retested on Linux + OS X, merging in.

**Pull request merged**